### DISCUSSION TITLE
Welcome to wallbox-mqtt-bridge Discussions!

### DISCUSSION THREAD

**Discussion by sweber** [sweber/wallbox-mqtt-bridge#1] [Announcements]
Feel free to discuss changes made here. Base functionality is from https://github.com/jagheterfredrik/wallbox-mqtt-bridge so keep in mind that general functionality should be discussed there.

**Comment by sincze** [comment]
> Feel free to discuss changes made here. Base functionality is from https://github.com/jagheterfredrik/wallbox-mqtt-bridge so keep in mind that general functionality should be discussed there.

Noticed your work, and tried to integrate it.

Maybe the A, B ... setting is a nice addition to the bridge as this is part of the official OCPP protocol. Now I need to do the translation in Node-Red...

I also noticed:


```
[lp-1  ] WARN 2024/10/28 19:24:48 deprecated: mincurrent setting is ignored, please remove
[lp-1  ] WARN 2024/10/28 19:24:48 deprecated: maxcurrent setting is ignored, please remove
[lp-1  ] WARN 2024/10/28 19:24:48 deprecated: phases setting is ignored, please remove

```

**Reply by sweber** [reply]
Yeah you're right. I got to admit that i copied that from my neighbor since I have a shelly pro 3em as energy meter infront of the wallbox.

I will remove it from the readme.

The a,b,c thing is a very evcc specific thing. Doesnt has to do something with the wallbox itself.  The combined source from evcc fit very well in this case. However i admit that it isn't the perfect way and obfuscates the functionality a bit.
I'll sleep over it :-)

Let me know if you have any problems getting it to run.

**Reply by sincze** [reply]
> Yeah you're right. I got to admit that i copied that from my neighbor since I have a shelly pro 3em as energy meter infront of the wallbox.

Stupid me did not integrate the 3EM during installation of the Wallbox ;-)
However due to the bridge I still have the LOADBALANCER data available so all is not lost here. 

I found this table that describes the A-F section.

```
Zustand | Fzg. angeschlossen | Laden möglich | Spannung CP-PE | Frequenz
------- | ------------------ | ------------- | -------------- | ----------
A       | Nein               | Nein          | 12V            |  DC
B       | Ja                 | Nein          | 9V/-12V        | 1 kHz
C       | Ja                 | Ja            | 6V/-12V        | 1 kHz
D       | Ja                 | Ja            | 3V/-12V        | 1 kHz
E       | Ja                 | Nein          | 0V             | -
F       | Ja                 | Nein          | -              | -
```

**Reply by sweber** [reply]
I just had a deeper look into the control pilot and the defined states. Fun fact: the bridge already has the control pilot topic. However that state is a combination of a hex-string represented as integer and the written out name of the state.
That is not perfect. I'll think splitting it up into seperate topics and cleaning up the state.

See https://github.com/sweber/wallbox-mqtt-bridge/blob/main/app/wallbox/wallbox_const.go
```go
var controlPilotStates = map[int]string{
	0xE:  "Error",
	0xF:  "Failure",
	0xA1: "Ready 1", // S1 at 12V, car not connected
	0xA2: "Ready 2",
	0xB1: "Connected 1", // S1 at 9V, car connected not allowed charge
	0xB2: "Connected 2", // S1 at Oscillator, car connected allowed charge
	0xC1: "Charging 1",
	0xC2: "Charging 2", // S2 closed
}
```

**Reply by sweber** [reply]
I'll will clean this up and provide 3 topics
```
control_pilot: (A,B,C,E,F)
control_pilot_detail: (A1, A2, B1, B2, C1, C2, E, F)
control_pilot_detail_message: (Ready 1, Ready 2, ...)
```
With that change no more weird mapping in the config is necessary and the control pilot sends the state as specified in DIN EN 61851-1:201

**Reply by sweber** [reply]
Release is done:
https://github.com/sweber/wallbox-mqtt-bridge/releases/tag/v20241029_1

**Comment by sincze** [comment]
:) can we add this as official PR as it matches all official documents?

**Reply by sweber** [reply]
I mean i can try. My last pull request was sadly denied by the original author. 🤷‍♂️ 😞

**Comment by sincze** [comment]
A but that was because of the time-out thing. This enhances the bridge with
official OCPP commands ensuring it is able to integrate with evcc.

**Reply by sweber** [reply]
Dont get me wrong, i'm not trying to lecture you. However it has nothing to do with OCPP. Its part of the specification how chargers have to communicate with the car over type 2 connectors.

And the Pulsar Plus exposes those information on the redis. However the bridge seems to do a mapping and doesn't provide it as the specification states it.

**Reply by sincze** [reply]
No prob ;-) 
I sincerely appreciate the effort.. I like to learn all the time. 

As I went through the evcc docs I noticed... Wallbox intregration via OCPP. And as that worked. I figured out... it must have something to do with it. As well as all the information that I could find on the internet.

**Comment by sincze** [comment]
https://docs.evcc.io/en/docs/devices/chargers#generic-support

this seems necessary ;-)

**Comment by Leventionz** [comment]
Hello, I will add my comment here as this seems the logical place to discuss.

I do not use EVCC, rather I would like to know if it is possible to control the control pilot states using home assistant.

For example, when my car is connected but I do not want it to charge as the electric price is high I could set state to 0xB1/0xB2 and when it is a cheap period I could set it to 0xC1/0xC2. Building the home assistant side of things is not the hard part it is understanding if it is possible to set the control pilot state remotely and have this persist on the wallbox.

**Reply by sweber** [reply]
Setting the control pilot states won't yield you the desired result.
The control pilot is an internal state for the Mode 2 Electric Vehicle Supply Equipment (EVSE) when they are connected with a Type 2 coupling. 
So those states are basically read only since the wallbox must be compliant to that specification.

However you can enable and disable charging by using the mqtt topic `wallbox_xxxxxx/charging_enable/set` with 0/1 as value. That will start or stop the charging.

For more information on the control pilot:
https://www.picoauto.com/library/automotive-guided-tests/charger-vehicle-communications-type-2

**Reply by Leventionz** [reply]
> Setting the control pilot states won't yield you the desired result. The control pilot is an internal state for the Mode 2 Electric Vehicle Supply Equipment (EVSE) when they are connected with a Type 2 coupling. So those states are basically read only since the wallbox must be compliant to that specification.
> 
> However you can enable and disable charging by using the mqtt topic `wallbox_xxxxxx/charging_enable/set` with 0/1 as value. That will start or stop the charging.

Thank you for the quick reply, must we use EVCC to use this mqtt topic?

**Reply by sweber** [reply]
As @sincze said. This bridge is totally unrelated to evcc. I just forked it to make it work with evcc flawlessly. Just use it with HA mqtt integration

**Comment by sincze** [comment]
Evcc not necessary.  Mqtt can be set with any mqtt client.

**Comment by gvisie** [comment]
Since the wallbox-mqtt-bridge from jagheterfredrik got updated to version 6.7.x , can this EVCC bridge get an update to?

**Reply by sincze** [reply]
would be even better if @jagheterfredrik would accept this EVCC addon functionality ;-) so we have 1 place to go!

**Reply by sweber** [reply]
I'm sorry,
But i will not working on this anymore. My pulsar plus gave up on me a year ago. Wallbox also made me an offer that i had to decline since they didn't want to repair it just replace it.

I switched to a better wallbox with automated phase switching and such.

But feel free to fork and adapt it.

**Reply by jagheterfredrik** [reply]
I do understand your need but not sure anything changed since https://github.com/jagheterfredrik/wallbox-mqtt-bridge/pull/65
